WalletWasabi/Services/TransactionProcessor.cs

```diff
@@ -23,15 +24,16 @@ public class TransactionProcessor
 
 		public event EventHandler<SmartCoin> CoinReceived;
 
-		public TransactionProcessor(KeyManager keyManager,
+		public TransactionProcessor(
+			AllTransactionStore transactionStore,
+			KeyManager keyManager,
 			ObservableConcurrentHashSet<SmartCoin> coins,
-			Money dustThreshold,
-			ConcurrentHashSet<SmartTransaction> transactionCache)
+			Money dustThreshold)
 		{
+			TransactionStore = Guard.NotNull(nameof(transactionStore), transactionStore);
 			KeyManager = Guard.NotNull(nameof(keyManager), keyManager);
 			Coins = Guard.NotNull(nameof(coins), coins);
 			DustThreshold = Guard.NotNull(nameof(dustThreshold), dustThreshold);
-			TransactionCache = Guard.NotNull(nameof(transactionCache), transactionCache);
```

Comment:
`transactionStore` cannot be `null`, should check with `Guard`

Reply:
What do you mean? It's checked.

Reply:
umm? I didn't see it... you're right.